credentials/alts/internal/handshaker/handshaker.go

```diff
@@ -92,12 +90,8 @@ func acquire() bool {
 func release() {
 	mu.Lock()
 	// If we need n to be configurable, we can pass it as an argument.
-	n := int64(1)
+	n := uint64(1)
 	concurrentHandshakes -= n
-	if concurrentHandshakes < 0 {
-		mu.Unlock()
-		panic("bad release")
-	}
 	mu.Unlock()
 }
```

Comment:
Is it possible to use a weighted semaphore type provided by the `sync` package, which does exactly what you want in here. See: https://pkg.go.dev/golang.org/x/sync/semaphore.

Reply:
Great suggestion! Done.